### PR TITLE
Posts and profile features

### PR DESCRIPTION
**Pull request by surbhi84**
This branch contains of the following features

- Consists of a single post page where you can see comments.
- User can add comments and delete post.
- Added profile page in this pr you can open user profiles by clicking on user names and images.
- Follow unfollow feature.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,28 +1,38 @@
...
+            (res) => res.data.user
+          );
+
+          dispatch(updateUserState({ encodedToken: token, user }));
+          const posts = await getPosts().then((res) => res.data.posts);
+          dispatch(setPosts(posts));
         }
       } catch (err) {
         console.error(err);
+      } finally {
```
Comment (HeyNitin) [inline src/App.tsx:34]: Good use of try-catch and finally block

File: src/App.tsx [hunk 2]
```diff
@@ -45,9 +55,15 @@ function App() {
 
   return (
     <div className={`App flex flex-col h-full w-full `}>
-      <Link to="/test">twopi</Link>
-      <Header />
-      <AppRoutes />
+      {isInitialRender ? (
+        <p>LOADING</p>
+      ) : (
+        <>
```
Comment (HeyNitin) [inline src/App.tsx:61]: Good use of fragments

**Review by HeyNitin — APPROVED** [review]
The code looks clean, you can add preview links or screenshots in descriptions.
Since there are no conflicts you can go ahead and merge

File: src/App.tsx (hunk 1 above)
Comment (shashikantmaurya0007) [inline src/App.tsx:34]: you can do initial call in index.js intself if you are using. redux you can do store.dispatch(andyour function here)

File: src/components/Sidebar.tsx [hunk 3]
```diff
@@ -53,8 +56,8 @@ export const Sidebar = () => {
         to="/bookmarks"
         className={({ isActive }) =>
           isActive
-            ? " flex flex-row items-center md:gap-3 gap-1 text-xl px-3 py-2 bg-primaryLight rounded-md dark:bg-darkLight "
-            : " flex flex-row items-center md:gap-3 gap-1 text-xl  px-3 py-2  "
+            ? " flex flex-row items-center md:gap-3 gap-1 px-3 py-2 bg-primaryLight rounded-md dark:bg-darkLight "
+            : " flex flex-row items-center md:gap-3 gap-1 px-3 py-2 hover:bg-primaryLight dark:hover:bg-darkLight rounded-md "
```
Comment (shashikantmaurya0007) [inline src/components/Sidebar.tsx:60]: better to extract this into a separate function as you are using it few more times

File: src/pages/PostPage.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                      Edit
+                      <FiEdit2 />
+                    </span>
+                    <span className="bg-darkCol h-[0.1rem]"></span>
+                    <span
+                      className="flex items-center mx-1 hover:scale-105 dark:hover:bg-darkCol dark:bg-darker p-1 rounded-md "
+                      onClick={() => {
+                        setShowOptions(false);
+                        setShowDeletePopup(true);
+                      }}
```
Comment (shashikantmaurya0007) [inline src/pages/PostPage.tsx:79]: try to avoid using logic in jsx

**Review by shashikantmaurya0007 — COMMENTED** [review]
brave of you to use typescript in major project.  suggested few changes 
i like your folder structure.
looks good to be merged

**Review by shashikantmaurya0007 — APPROVED** [review]

**Pull request merged**